sdks/java/core/src/main/java/org/apache/beam/sdk/runners/inprocess/ExecutorServiceParallelExecutor.java

```diff
@@ -269,25 +273,38 @@ public void handleThrowable(CommittedBundle<?> inputBundle, Throwable t) {
    */
   private static class ExecutorUpdate {
     private final Optional<? extends CommittedBundle<?>> bundle;
+    private final Optional<? extends AppliedPTransform<?, ?, ?>> consumer;
```

Comment:
Comment here about the invariant that this is either/or where two things are null together.